pkg/aws/eni/node_manager.go

```diff
@@ -119,9 +119,34 @@ func NewNodeManager(instancesAPI nodeManagerAPI, ec2API ec2API, k8sAPI k8sAPI, m
 
 	mngr.resyncTrigger = resyncTrigger
 
+	// Assume readiness, the initial blocking resync in Start() will update
+	// the readiness
+	mngr.SetInstancesAPIReadiness(true)
+
 	return mngr, nil
 }
 
+func (n *NodeManager) InstancesAPIResync(ctx context.Context) (time.Time, bool) {
+	syncTime := n.instancesAPI.Resync(ctx)
+	success := !syncTime.IsZero()
+	n.SetInstancesAPIReadiness(success)
+	return syncTime, success
+}
+
+// SetInstancesAPIReadiness sets the readiness state of the instances API
+func (n *NodeManager) SetInstancesAPIReadiness(ready bool) {
+	n.mutex.Lock()
+	n.stableInstancesAPI = ready
+	n.mutex.Unlock()
+}
+
+// InstancesAPIIsReady returns true if the instances API is stable and ready
+func (n *NodeManager) InstancesAPIIsReady() bool {
```

Comment:
Where did the user of this function go?

Reply:
Great catch! Adding this to last commit.